src/tcgwars/logic/impl/gen4/DiamondPearl.groovy

```diff
@@ -825,17 +988,30 @@ public enum DiamondPearl implements LogicCardInfo {
           move "Healing Wish", {
             text "Flip a coin until you get tails. Remove a number of damage counters equal to the number of heads from 1 of your Pokémon."
             energyCost C
-            attackRequirement {}
+            attackRequirement {
+              assert my.all.findAll { it.numberOfDamageCounters }: "No damaged Pokemon"
+            }
             onAttack {
-              damage 0
+              healAmount = 0
+              flipUntilTails { healAmount += 10 }
+              if (healAmount && my.all.findAll { it.numberOfDamageCounters }) {
+                def pcs = my.all.findAll { it.numberOfDamageCounters }.select()
+                heal healAmount, pcs
+              }
             }
           }
           move "Flop", {
             text "30 damage. You may switch Lopunny with 1 of your Benched Pokémon."
             energyCost C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 30
+              //TODO: This could definitely be generalized into a method, there's a couple dozens of cards with this optional effect across the DPPt era.
```

Comment:
`switchYourActive()` like Archeops in UnifiedMinds but with a `confirm`? You can probably add something slightly altered to `TcgStatics.groovy`

Reply:
Aha! It seems to be implemented already in that command itself, thanks a lot <3